packages/react-components/src/components/GridLayout.tsx

```diff
@@ -31,42 +24,42 @@ export interface GridLayoutProps {
   styles?: BaseCustomStylesProps;
 }
 
-const calculateStandardLayoutRows = (numberOfItems: number, gridCol: number): number =>
-  Math.ceil(numberOfItems / gridCol);
-
-const calculateStandardLayoutColumns = (numberOfItems: number): number =>
-  numberOfItems > 0 ? Math.ceil(Math.sqrt(numberOfItems)) : 1;
-
 /**
- * A component to lay out audio / video tiles in a call.
+ * A component to lay out audio / video participants tiles in a call.
  *
  * @public
  */
 export const GridLayout = (props: GridLayoutProps): JSX.Element => {
-  const [gridCol, setGridCol] = useState(1);
-  const [gridRow, setGridRow] = useState(1);
-
-  const { children, layout = 'standard', styles } = props;
+  const { children, styles } = props;
   const numberOfChildren = React.Children.count(children);
 
-  switch (layout) {
-    case 'standard': {
-      const numberOfColumns = calculateStandardLayoutColumns(numberOfChildren);
-      if (gridCol !== numberOfColumns) setGridCol(numberOfColumns);
-      const numberOfRows = calculateStandardLayoutRows(numberOfChildren, gridCol);
-      if (gridRow !== numberOfRows) setGridRow(numberOfRows);
-      break;
+  const containerRef = useRef<HTMLDivElement>(null);
+  const [gridProps, setGridProps] = useState<GridProps>({
+    horizontalFill: true,
+    rows: Math.ceil(Math.sqrt(numberOfChildren)),
+    columns: Math.ceil(Math.sqrt(numberOfChildren))
+  });
+
+  useEffect(() => {
+    const updateDynamicGridStyles = (): void => {
+      if (containerRef.current) {
+        setGridProps(
+          calculateGridProps(numberOfChildren, containerRef.current.offsetWidth, containerRef.current.offsetHeight)
+        );
+      }
+    };
+    const observer = new ResizeObserver(updateDynamicGridStyles);
+    if (containerRef.current) {
+      observer.observe(containerRef.current);
     }
-  }
+    updateDynamicGridStyles();
+    return () => observer.disconnect();
+  }, [numberOfChildren, containerRef.current?.offsetWidth, containerRef.current?.offsetHeight]);
```

Comment:
I like that we only recalculate the layout based on the number of children so if we start doing things like dominant speaking where they may change position but the length is the same, we can reuse the exact layout